app/services/description_headers.rb

```diff
@@ -11,14 +11,20 @@ def initialize(headers:)
   end
 
   def create
-    sort_order = %w[title contributor form event language note identifier access purl subject relatedResource adminMetadata geographic marcEncodedData valueAt]
-    updated_headers = (@headers - ['source_id']).sort
-    ['source_id'] + updated_headers.sort_by.with_index { |header, index| [sort_order.index(field_name(header)) || headers.length, index] }
+    headers
+      .sort_by
+      .with_index { |header, index| [sort_order.index(field_name(header)) || headers.length, index] }
   end
 
   private
 
+  attr_reader :headers
+
   def field_name(key)
-    /[[:alpha:]]*/.match(key)[0]
+    /([[:alpha:]]|_)*/.match(key)[0]
+  end
+
+  def sort_order
+    %w[source_id title contributor form event language note identifier access purl subject relatedResource adminMetadata geographic marcEncodedData valueAt]
```

Comment:
Instead of mucking about with the `source_id` in the headers (old lines 15-16), specify `source_id` in the sort order.